applications/ThermalDEMApplication/custom_elements/thermal_spheric_particle.cpp

```diff
@@ -0,0 +1,1443 @@
+    if (mPrescribedHeatFluxSurface != 0.0)
+      mPrescribedHeatFlux += mPrescribedHeatFluxSurface * GetParticleSurfaceArea();
+
+    // Prescribed heat source over volume
+    if (mPrescribedHeatFluxVolume != 0.0)
+      mPrescribedHeatFlux += mPrescribedHeatFluxVolume * GetParticleVolume();
+
+    // Sum up heat fluxes contributions
+    mTotalHeatFlux = mConductionDirectHeatFlux + mConductionIndirectHeatFlux + mRadiationHeatFlux + mFrictionHeatFlux + mConvectionHeatFlux + mPrescribedHeatFlux;
+    SetParticleHeatFlux(mTotalHeatFlux);
+
+    KRATOS_CATCH("")
+  }
+
+  //------------------------------------------------------------------------------------------------------------
+  void ThermalSphericParticle::ComputeHeatFluxWithNeighbor(const ProcessInfo& r_process_info) {
+    KRATOS_TRY
+
+    // Check if neighbor is adiabatic
+    if (CheckAdiabaticNeighbor())
+      return;
+
+    // Compute simulated or adjusted interaction properties
+    ComputeInteractionProps(r_process_info);
+
+    // Heat transfer mechanisms
+    if (this->Is(DEMThermalFlags::HAS_DIRECT_CONDUCTION))
+      mConductionDirectHeatFlux += GetDirectConductionModel().ComputeHeatFlux(r_process_info, this);
+
+    if (this->Is(DEMThermalFlags::HAS_INDIRECT_CONDUCTION))
+      mConductionIndirectHeatFlux += GetIndirectConductionModel().ComputeHeatFlux(r_process_info, this);
+
+    if (this->Is(DEMThermalFlags::HAS_RADIATION))
+      mRadiationHeatFlux += GetRadiationModel().ComputeHeatFlux(r_process_info, this);
+
+    if (this->Is(DEMThermalFlags::HAS_FRICTION_HEAT) && this->Is(DEMThermalFlags::HAS_MOTION))
+      mFrictionHeatFlux += GetFrictionModel().ComputeHeatGeneration(r_process_info, this);
+
+    KRATOS_CATCH("")
+  }
+
+  //------------------------------------------------------------------------------------------------------------
+  void ThermalSphericParticle::ComputeInteractionProps(const ProcessInfo& r_process_info) {
+    KRATOS_TRY
+
+    // Set simulated properties
+    mNeighborDistance   = ComputeDistanceToNeighbor();
+    mNeighborSeparation = ComputeSeparationToNeighbor();
+    mNeighborInContact  = CheckSurfaceDistance(0.0);
+    mContactRadius      = ComputeContactRadius();
+
+    // Set adjusted contact properties
+    if (this->Is(DEMThermalFlags::HAS_REAL_CONTACT) && mNeighborInContact) {
+      GetRealContactModel().AdjustContact(r_process_info, this);
+    }
+    else {
+      mNeighborDistanceAdjusted   = mNeighborDistance;
+      mNeighborSeparationAdjusted = mNeighborSeparation;
+      mContactRadiusAdjusted      = mContactRadius;
+    }
+
+    KRATOS_CATCH("")
+  }
+
+  //------------------------------------------------------------------------------------------------------------
+  void ThermalSphericParticle::StoreBallToBallContactInfo(const ProcessInfo& r_process_info, SphericParticle::ParticleDataBuffer& data_buffer, SphericParticle* neighbor, double GlobalContactForce[3], bool sliding) {
+    KRATOS_TRY
+
```

Comment:
I would here call the base class method. Now it is empty, but it may not be in the future.

Reply:
Now calling empty base method.